lib/carrierwave/storage/azure_rm.rb

```diff
@@ -22,19 +22,34 @@ def connection
         end
       end
 
+      def signer
+        @signer ||= begin
+          ::Azure::Storage::Core::Auth::SharedAccessSignature.new
```

Comment:
Wouldn't it be better for `store!` method to have write permission, and `retrieve!` to have only read permission?